### DISCUSSION TITLE
Column Labels

### DISCUSSION THREAD

**Discussion by rnugent3** [HydrologicEngineeringCenter/HEC-FDA#313] [Q&A] [answered]
The user interface needs good column labels for all of our tables. For example, if a user is entering a stage-discharge function with uncertainty where the distributional assumption is triangular, then we want four columns: discharge, minimum stage, most likely stage, and maximum stage. What are the different ways to make this happen?

**Comment by Brennan1994** [comment, marked as answer]
The way I'd propose to handle this is as I did in the Pilot UI. 

![image](https://user-images.githubusercontent.com/64556723/163470393-68137570-91d4-47ca-a618-5b41fa0b7fd0.png)

I added a property to the ComputeComponentVM (which is a base level vm each of the relationships builds off of) which holds a string in the format shown above (xunit, yunit). I set that property on construction, and bind it to the view to display above the table with plot. This lets our tables stay as generic and reusable as possible, and keeps the changing of units out of more complicated data table code. 

Currently, column headers are set by attribute on the row classes, like in the triangularRow class here: 
![image](https://user-images.githubusercontent.com/64556723/163471528-6b255570-42f0-4069-9552-abadf811df05.png)

We could make this a dynamic title instead, and pass down information to it from it's creator so that attribute could be set as some dynamic string variable + "min" . It's just a little more complicated, and I found my solution acceptable. 

EDIT: on further investigation, we can't use a non static object in the constructor of an attribute, so it would require some finagling to get things working to display parameter names in the column headers. I can go down that route of investigation, but my preference is for my proposal above, and would like the team to make the decision that it is unacceptable before spending too much time experimenting.

**Reply by rnugent3** [reply]
I am good with your proposal. X Value and Y Value are OK. I think it is reasonable for an MVP. It helps to have the (X,Y) next to the name of the function. I am going to add column labels to our list of postponed bells and whistles.
